kitchen.yml

```diff
@@ -173,6 +173,19 @@ platforms:
     driver:
       image: netmanagers/salt-2017.7-py2:arch-base-latest
       run_command: /usr/lib/systemd/systemd
+<% if not ENV['CI'] == 'true' %>
+  - name: windows-81-2019-2-py3
+    driver:
+      name: vagrant
+      box: techneg/win81x64-pro-salt
```

Comment:
```suggestion
      box: techneg/win81x64-pro-salt
      customize:
        usbxhci: 'off'
      gui: false
```

* Ouch!  These three lines were discovered at some cost.

---

#### `gui: false`

ci(kitchen): use `headless` mode for Vagrant

* Avoid following error:

```bash
$ bin/kitchen converge ${INSTANCE}
  ...
  Stderr: VBoxManage: error: The machine 'kitchen-openvpn-formula-default-windows-81-2019-2-py3-5782a2d9-81b9-4bcd-ac6c-ff9a028a2413' is already locked by a session (or being locked or unlocked)
  VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MachineWrap, interface IMachine, callee nsISupports
  VBoxManage: error: Context: "LaunchVMProcess(a->session, sessionType.raw(), Bstr(strEnv).raw(), progress.asOutParam())" at line 600 of file VBoxManageMisc.cpp
```

---

#### `usbxhci: 'off'`

ci(kitchen): avoid using USB controller for Vagrant

* https://github.com/test-kitchen/kitchen-vagrant#-customize
* Avoid following error:

```bash
$ bin/kitchen converge ${INSTANCE}
  ...
  Stderr: VBoxManage: error: Implementation of the USB 3.0 controller not found!
  VBoxManage: error: Because the USB 3.0 controller state is part of the saved VM state, the VM cannot be started. To fix this problem, either install the 'Oracle VM VirtualBox Extension Pack' or disable USB 3.0 sup
  port in the VM settings (VERR_NOT_FOUND)
```

Reply:
I'll set these in the Vagrant box itself. It's a box I've used for local development for a while so I hadn't encountered these errors.

Reply:
@dafyddj That'd be good; even when I tried to modify the settings using VirtualBox, they got overwritten when running `kitchen converge`, before I worked out these settings.  I'd still prefer these to be included, to save trouble in the future when using other Vagrant boxes.

Reply:
I have left `gui: false` in as good practice, but don't believe it's necessary to leave the other customisation. The option in my original box was really niche, and shouldn't have been there.

Reply:
I'm fine either way.  My take on it is that it doesn't introduce any problems by including it; it's happened once, it could always take place again, if someone else provides Vagrant boxes in the future.  The errors encountered are frustrating enough to put one off from proceeding with the local testing; that's the key issue here -- to make testing as accessible as possible.

As I've said, I'm fine with this in this situation, so I'm marking this as resolved.